app/src/test/java/com/spascoding/taskycourse/feature_auth/presentation/util/NameInitialsExtractorTest.kt

```diff
@@ -0,0 +1,34 @@
+package com.spascoding.taskycourse.feature_auth.presentation.util
+
+import com.google.common.truth.Truth.assertThat
+import org.junit.Assert
+import org.junit.Test
+
+class NameInitialsExtractorTest {
+
+    @Test
+    fun `Extract form one word`() {
+        val oneWordName = "Spas"
+        val actual = NameInitialsExtractor(oneWordName).extract()
+        val expected = "SP"
+        assertThat(actual).isEqualTo(expected)
+    }
+
+    @Test
+    fun `Extract form multiple word`() {
+        val multipleWordName = "Spas Arnaudov"
+        val actual = NameInitialsExtractor(multipleWordName).extract()
+        val expected = "SA"
+        assertThat(actual).isEqualTo(expected)
+    }
+
+    @Test
+    fun `Extract form empty word`() {
+        val emptyWordName = ""
+        val exception = Assert.assertThrows(IllegalArgumentException::class.java) {
+            NameInitialsExtractor(emptyWordName).extract()
+        }
+        assertThat(exception).hasMessageThat().isEqualTo("Name must contains minimum two letters")
+    }
+
+}
```

Comment:
This misses some tests on edge cases like I've commented on above :) 

Also, a great way to enhance such tests that have a simple input and output are parameterized tests. You can write all your tests inside of one test that way. E.g. for validating passwords:
```
@ParameterizedTest
@CsvSource(
    "Test12345, true",
    "test12345, false",
    "12345, false",
    "Test-1234, true",
    "TEST12345, false",
)
fun testValidatePassword(password: String, expectedIsValid: Boolean) {
    val state = userDataValidator.validatePassword(password)

    assertThat(state.isValidPassword).isEqualTo(expectedIsValid)
}
```
You need to use JUnit5 for that however. It will run for each input and output once.